### ISSUE TITLE
CLI arg to ignore "workspace:" prefix and install from remote

### ISSUE THREAD

**Issue by onlywei** [pnpm/pnpm#8898]
### Contribution

- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

During local development, I want pnpm install to link all of the packages in the monorepo that are marked as workspace:. However, during CI I would rather download those packages from my internal package registry (or cached from store) so that I can avoid building those packages in CI.

Is this possible?

### Describe the solution you'd like

I'd like a CLI argument such as the one below that will install all dependencies in package.json that are prefixed with "workspace:" as if they didn't have that prefix.

```
pnpm install --ignore-workspace
```

### Describe the drawbacks of your solution

I think this may introduce a branching path in the pnpm-lock.yaml file for each workspace package. During normal installs, we want to locally link the packages, but during `--ignore-workspace` installs, we should still somehow lock the result into the pnpm-lock.yaml file.

### Describe alternatives you've considered

I thought about writing a script on my end to manually remove the "workspace:" prefixes from my package.json files, but that can get quite annoying to implement in different environments (in docker vs out of docker).

**Comment by zkochan** [comment]
It sounds like you need something like Turborepo's remote caching. Their competitors have similar features.

[Bit cloud](https://bit.cloud/) does this out of the box.

On our end I guess we could have something similar to Turborepo's remote caching. However, we might not be able to match turborepo's performance as it is written in Rust

**Comment by onlywei** [comment]
@zkochan I don't really want something as complex as Turborepo/NX remote caching. I just want to be able to use a CLI arg to instruct pnpm whether to install via linking locally or from the registry. When installing from the registry, it doesn't even have to download from the internet -- if the version is already in the pnpm store we can still use that.

**Comment by HJZzzzz** [comment]
this also a valid use case for me, any updates?

**Comment by zkochan** [comment]
Instead of installing from remote into node_modules, I would rather add a way to fetch the compiled artifacts and add them to the "dist" directories. If we fetch these from the registry, then it will only work for those packages in the workspace that were not modified since last release.

**Comment by HJZzzzz** [comment]
> Instead of installing from remote into node_modules, I would rather add a way to fetch the compiled artifacts and add them to the "dist" directories. If we fetch these from the registry, then it will only work for those packages in the workspace that were not modified since last release.

agree, but isn't it replicating a remote cache kind of thing？😂 installing from registry seems much more easier with less effort right?

**Comment by zkochan** [comment]
It doesn't seem easier for me to install from registry because we'd have to change the lockfile. If we don't change the lockfile, then it will be out-of-sync with node_modules after running `pnpm install --ignore-workspace`. Remote cache seems to work fine as several monorepo managing tools implemented it. No need to reinvent the wheel I guess.

**Comment by HJZzzzz** [comment]
you are right, seems remote cache is a better solution😂 thanks~

**Comment by onlywei** [comment]
> It doesn't seem easier for me to install from registry because we'd have to change the lockfile. If we don't change the lockfile, then it will be out-of-sync with node_modules after running `pnpm install --ignore-workspace`. Remote cache seems to work fine as several monorepo managing tools implemented it. No need to reinvent the wheel I guess.

Yes it does seem like the lockfile would have to contain some data such as IF local use local link, otherwise install this specific version from the store. Definitely not a trivial change. I still think it could be quite valuable though since Nx and Turborepo are pretty heavyweight solutions that people may want to avoid.
